merkle/merkle.ts

```diff
@@ -55,35 +67,31 @@ const sortAndAddIndex = (balances) => {
   });
 
   return balances.map(function (a, i) {
-    return { address: a, balance: balances[a], index: i };
+    return { ...a, index: i };
   });
 }
 
 
-const computeRoot = () => {
-    let root = []
-	let nextHash = []
-	const leaves = sortAndAddIndex(data);
-    const leaf = leaves[0]
-	console.log({leaves})
-    const leafIndex = leaves.indexOf(leaf)
-
-
-    let {firstLevel, adjacentHash} =  firstHashing(leaves, leafIndex)
-    const firstHash = adjacentHash
-    root.push(...firstLevel)
-    while (root.length > 1) {
-        const {nextLevel, returnedNextHash} =  reduceMerkleBranches(root, adjacentHash)
-        if (returnedNextHash) {
-            nextHash.push(returnedNextHash)
-            adjacentHash = returnedNextHash
-        }
-        root.push(...nextLevel)
-	}
-	nextHash.unshift(firstHash)
-    console.log("proof", {nextHash, root: root[0], leaf})
-    console.log({root})
+const computeRoot = (balances) => {
+    const leaves = sortAndAddIndex(balances);
+    let hashedLeaves: string[] = leaves.map(hash);
+
+    while (hashedLeaves.length > 1) {
+        hashedLeaves = reduceMerkleBranches(hashedLeaves);
+    }
+
+    return hashedLeaves[0];
+}
 
+const computeAllProofs = (balances) => {
+    const leaves = sortAndAddIndex(balances);
+    let proofs = {};
+    leaves.forEach(leaf => {
+        proofs[leaf.address] = computeMerkleProof(leaves, leaf.index);
+        writeToFileSystem(JSON.stringify(proofs[leaf.address]), leaf.address);
```

Comment:
idk if we need this, this will be heavy to run locally while testing and kinda annoying. I think this would be best to write to all proofs and when we go to upload it to S3 just do a map for each entry and use the address as a key, cleaner in the long run I think.

Reply:
added 52ae30d